java/libs/src/main/java/org/physical_web/collection/PhysicalWebCollection.java

```diff
@@ -245,6 +245,39 @@ public void jsonDeserialize(JSONObject jsonObject) throws PhysicalWebCollectionE
    */
   public List<PwPair> getPwPairsSortedByRank() {
     // Get all valid PwPairs.
+    List<PwPair> allPwPairs = getPwPairs();
```

Comment:
In our abstraction UrlDevice represents the beacon and PwsResult represents the cached URL metadata from PWS.  The problem with attaching UrlDevice to PwsResult is that multiple devices could be broadcasting the same URL, and we only keep one PwsResult per broadcast URL.  Instead of one UrlDevice reference the PwsResult would need to keep a list and pick the correct one based on rank.  Maintaining this list as beacons are found and lost is likely more complex than building the associations and filtering them at display time as we do now.

(btw, we actually do store the initial URL of the UrlDevice in PwsResult as mRequestUrl.)

I think there's value in keeping the PWS-derived metadata and beacon-derived metadata logically separated.  PwPair does serve a purpose in providing a place to attach behaviors (like ordering) that depend on both the device and URL metadata.